source/keyboardHandler.py

```diff
@@ -217,6 +222,7 @@ def internal_keyDownEvent(vkCode,scanCode,extended,injected):
 			and not gestureExecuted 
 			# and not if this gesture is a modifier key
 			and not isNVDAModifierKey(vkCode,extended) and not vkCode in KeyboardInputGesture.NORMAL_MODIFIER_KEYS
+			and not watchdog.isAttemptingRecovery

```

Comment:
I would prefer to see this as its own check returning true above this if condition, as I fear it gets lost otherwise.